underfs/s3/src/main/java/tachyon/underfs/s3/S3UnderFileSystem.java

```diff
@@ -347,6 +333,23 @@ private boolean deleteInternal(String key) {
   }
 
   /**
+   * Gets the child name based on the parent name. If the parent is not valid, the entire child
```

Comment:
"If parent is not valid, the entire child key will be returned."
If parent is not valid (i.e., not the prefix of child), isn't this an error case in our code path?
If true, we shouldn't simpley return the original child string (with a warning though). Instead, we should escalate the error to the caller. e.g., by returning "".